### PR TITLE
add request logging and hooking

### PR DESCRIPTION
**Pull request by tysonholub**
PR adds request/response logging and request hooks to TwilioHttpClient

### CODE REVIEW

File: twilio/http/http_client.py [hunk 1]
```diff
@@ -1,20 +1,30 @@
...
+from requests import Request, Session, hooks
 
 from twilio.http import HttpClient, get_cert_file
 from twilio.http.response import Response
+import logging
+import json
+try:
+    from urllib.parse import urlencode
+except ImportError:
+    from urllib import urlencode
```
Comment (efossier) [inline twilio/http/http_client.py]: you should import things via the `compat.py` module which handles version specific imports like this for you.

**Comment by tysonholub** [comment]
pushed commit to change urlencode import from twilio.compat

**Comment by jingming** [comment]
@tysonholub Thank you for your PR! We have made a ticket in our internal tracking system and will get to this as soon as we can.

File: twilio/http/http_client.py [hunk 2]
```diff
@@ -47,4 +72,12 @@ def request(self, method, url, params=None, data=None, headers=None, auth=None,
             timeout=timeout,
         )
 
+        if hasattr(response, 'json') and response.json():
+            # strip JSON of possible prettyprint
```
Comment (dougblack) [inline twilio/http/http_client.py]: The API doesn't pretty print. We probably want to save the `loads`/`dumps` cost by just always using `response.text`.

File: twilio/http/http_client.py [hunk 3]
```diff
@@ -38,7 +45,25 @@ def request(self, method, url, params=None, data=None, headers=None, auth=None,
             session = Session()
             session.verify = get_cert_file()
 
-        request = Request(method.upper(), url, params=params, data=data, headers=headers, auth=auth)
+        kwargs = dict(
```
Comment (dougblack) [inline twilio/http/http_client.py]: nit: Let's just use a dict literal here.

**Comment by tysonholub** [comment]
I should be able to address these issues before Friday. Thanks!

**Pull request merged**